tests/run-make/emit-to-stdout/rmake.rs

```diff
@@ -0,0 +1,60 @@
+// Test that emitting to stdout works
+
+use std::fs::{self, File};
+
+use run_make_support::{diff, run_in_tmpdir, rustc};
+
+// Test emitting text outputs to stdout works correctly
+fn run_diff(name: &str, file_args: &[&str]) {
+    rustc().emit(format!("{name}={name}")).input("test.rs").args(file_args).run();
+    let out = rustc().emit(format!("{name}=-")).input("test.rs").run().stdout_utf8();
+    diff().expected_file(name).actual_text("stdout", &out).run();
+}
+
+// Test that emitting binary formats to a terminal gives the correct error
+fn run_terminal_err_diff(name: &str) {
+    let pty = File::create("/dev/ptmx").unwrap();
+    let err = File::create(name).unwrap();
```

Comment:
Problem: this will only work for unix and msys2, but not native windows.

@ChrisDenton came up with a version that might work for native Windows in https://github.com/rust-lang/rust/pull/128460#issuecomment-2380716435, reproduced below:

```rs
#[cfg(not(windows))]
let stdout = std::fs::File::options().write(true).open("/dev/ptmx").unwrap();
// FIXME: If this test fails and the compiler does print to the console, then this will produce a lot of output.
// We should spawn a new console instead to print stdout.
#[cfg(windows)]
let stdout = std::fs::File::options().read(true).write(true).open(r"\\.\CONOUT$").unwrap();
```